### ISSUE TITLE
Spell-checking RST files doesn't skip custom labels

### ISSUE THREAD

**Issue by amyq** [vale-cli/vale#162]
We've been noticing lately that we're seeing spelling errors on custom labels, which are directives in RST. 

For example:

```
.. _ac-sshkey-add:

ac-sshkey-add
^^^^^^^^^^^^^

Add an SSH key to a website.
```

The subheading should absolutely be flagged, because `ac-sshkey-add` is not a dictionary word, but `.. _ac-sshkey-add:` is a RST directive (custom label).

Any suggestions on how to get around this problem?

- OS: macOS
- Install method: homebrew
- Vale version: 1.7.1

**Comment by CashWilliams** [comment]
Adding:

```
filters:
  - '\.\.\s\b_\S*:\n'
```

or even

```
filters:
  - '\.\..*\n'
```

to the `Spelling.yml` file does not fix the problem. But it does feel like there is a larger issue in that all RST directives should be skipped for spelling.

**Comment by jdkato** [comment]
There have been no recent changes to how reStructredText is handled, so any issues you're experiencing aren't new. My only suggestion at the moment is to wait for #147 to be resolved (see #145 and #146 for more information).

Also, all markup formats are converted to HTML (which can be traversed in useful ways -- e.g., ignoring code blocks) prior to applying any patterns. This means that `filters` can't match markup syntax itself -- only the actual text content of the file.

**Comment by jdkato** [comment]
This should be fixed in [`v2.0.0-beta.3`](https://github.com/errata-ai/vale/releases/tag/v2.0.0-beta.3).
